### ISSUE TITLE
HID device not working on Debian image (Kernel 4.9.80)

### ISSUE THREAD

**Issue by superjcvd** [qlyoung/armory-keyboard#6]
Hello,

First of all thanks a lot for your work and for sharing it.
I tried to use your hid-ecm.sh on a Debian image on Usb Armory but I can't make it work at all.
https://github.com/inversepath/usbarmory-debian-base_image/releases

I tried the acm serial port gadget (seems to work)
I tried the RNDIS gadget (works fine with Windows)
I tried the ECM gadget
But when trying to activate the HID gadget I can't see anything detected on the USB of the guest computer.

Maybe you can help me ?
I can provide the logs and the different .sh scripts I tried to execute so far

regards

**Comment by qlyoung** [comment]
Yes, logs would be helpful. Can you also show the steps (shell commands) you're taking to set it up?

**Comment by superjcvd** [comment]
Hello,

After investigation it seems that the problem comes from the Kernel 4.9 in Debian ARM.
After switching to kernel 4.14 everything is working great !!!

By the way I created a layout for French Azerty Keyboard if you need it just ask me

**Comment by qlyoung** [comment]
Cool, glad it's working for you :) And yes, if you want to submit the layout in a PR, or paste it somewhere, I'll merge it in. Thanks!

**Comment by superjcvd** [comment]
Here it is

French AZERTY layout (not fully tested but so far seems good)
[french.layout.txt](https://github.com/qlyoung/armory-keyboard/files/1748683/french.layout.txt)

**Comment by superjcvd** [comment]
Please find the script file that I use in order to activate the HID gadget + Network card
Under Windows 10 ECM driver is not working. I add to replace it by RNDIS instead

[myusbgadget.sh.txt](https://github.com/qlyoung/armory-keyboard/files/1748685/myusbgadget.sh.txt)

**Comment by qlyoung** [comment]
Layout has been merged and I'll leave this open until I have time to test your script on Windows. Thanks!

**Comment by superjcvd** [comment]
thanks to you :)
